app/validators/routing/types.py

```diff
@@ -42,33 +42,67 @@
     "NoneType": TYPE_NULL,
 }
 
+METADATA_TYPE_TO_JSON_TYPE = {
+    "string": TYPE_STRING,
+    "date": TYPE_STRING,
+    "boolean": TYPE_BOOLEAN,
+}
+
+
+def resolve_answer_source_json_type(answer_id, schema):
+    answer_type = schema.answers_with_context[answer_id]["answer"]["type"]
+    return ANSWER_TYPE_TO_JSON_TYPE[answer_type]
+
+
+def resolve_calculated_summary_source_json_type(block_id, schema):
+    block = schema.get_block(block_id)
+    if block["calculation"].get("answers_to_calculate"):
+        answer_id = block["calculation"]["answers_to_calculate"][0]
+    else:
+        answer_value_source = block["calculation"]["operation"]["+"][0]
+        answer_id = answer_value_source["identifier"]
+    answer_type = schema.answers_with_context[answer_id]["answer"]["type"]
+    return ANSWER_TYPE_TO_JSON_TYPE[answer_type]
+
+
+def resolve_grand_calculated_summary_source_json_type(block_id, schema):
+    block = schema.get_block(block_id)
+    first_calculated_summary_source = block["calculation"]["operation"]["+"][0]
+    return resolve_value_source_json_type(first_calculated_summary_source, schema)
+
+
+def resolve_metadata_source_json_type(identifier, schema):
+    if identifier:
+        for values in schema.schema.get("metadata", []):
+            if values.get("name") == identifier:
+                return METADATA_TYPE_TO_JSON_TYPE[values.get("type")]
+    return TYPE_STRING
+
+
+def resolve_list_source_json_type(selector):
+    if selector:
+        return LIST_SELECTOR_TO_JSON_TYPE[selector]
+    return TYPE_ARRAY
```

Comment:
This could be simplified
```suggestion
return LIST_SELECTOR_TO_JSON_TYPE[selector] if selector else TYPE_ARRAY
```